catchup/service.go

```diff
@@ -152,7 +152,7 @@ func (s *Service) SynchronizingTime() time.Duration {
 
 // function scope to make a bunch of defer statements better
 func (s *Service) innerFetch(fetcher Fetcher, r basics.Round) (blk *bookkeeping.Block, cert *agreement.Certificate, rpcc FetcherClient, err error) {
-	ctx, cf := context.WithTimeout(s.ctx, DefaultFetchTimeout)
+	ctx, cf := context.WithCancel(s.ctx)
```

Comment:
Can you please explain why we don't need the timeout here?

Reply:
This timeout would automatically cancel the context. Content cancellation (naturally) is only meaningful when being examined. Concretely, it's being examined within the http and gossip fetchers themselves, where I have replaced it with a fetcher-specific timeout context. ( i.e. moving the global one into a fetcher-specific one, and also making it configurable )